src/main/json2xlsx.js

```diff
@@ -22,31 +24,33 @@ const run = async (args) => {
   const builder = new ccdUtils.SpreadsheetBuilder(sourceXlsx);
   await builder.loadAsync();
 
-  const excludedFilenamePatterns = args.excludedFilenamePatterns ? stringUtils.split(args.excludedFilenamePatterns) : [];
-  const files = fileUtils.listFilesInDirectory(args.sheetsDir, excludedFilenamePatterns);
+  const excludedFilenamePatterns = args.excludedFilenamePatterns ?
+    stringUtils.split(args.excludedFilenamePatterns) : [];
 
-  for (const file of files) {
-    const readSheetData = async (file) => {
-      const readJsonFile = (relativeFilePath) => {
-        return fileUtils.readJson(path.join(args.sheetsDir, relativeFilePath), Substitutor.injectEnvironmentVariables);
-      };
+  const paths = await fileUtils.getJsonFilePaths(args.sheetsDir, excludedFilenamePatterns);
+  const fileMap = await sheetUtils.groupToSheets(paths);
 
-      if (file.isDirectory()) {
+  for (const file in fileMap) {
+    const readSheetData = async (filesFragments) => {
+      const readJsonFile = (relativeFilePath) => fileUtils.readJson(
+        path.join(args.sheetsDir, relativeFilePath), Substitutor.injectEnvironmentVariables);
+
+      if (filesFragments.length === 0) {
+        return readJsonFile(file);
+      } else {
         const jsonFragments = await Promise.all(
-          fileUtils.listFilesInDirectory(path.join(args.sheetsDir, file.name), excludedFilenamePatterns)
-            .map(fragmentFile => readJsonFile(`${file.name}/${fragmentFile.name}`))
-        );
+          filesFragments.map(fileFragment => readJsonFile(`${file}/${fileFragment}`)));
         return jsonFragments.flat();
-      } else {
-        return await readJsonFile(file.name);
       }
     };
 
-    console.log(`  importing sheet data from ${file.name} ${file.isDirectory() ? 'directory' : 'file'}`);
-    const json = await readSheetData(file);
+    console.log(`  importing sheet data from ${file} ${file.indexOf('.json') === -1 ? 'directory' : 'file'}`);
+
+    const fragments = fileMap[file].length === 0 ? [] : fileMap[file];
+    const json = await readSheetData(fragments);
```

Comment:
That is so confusing. Function accepts array of fragments but for top level JSON file an application passes empty array. Here is confusing part. Despite of empty array being passed function still reads data from file defined in outer scope.

Please refactor so that it is clear what is function's input and output.